crates/shielded_token/src/masp/shielded_wallet.rs

```diff
@@ -1181,6 +1200,50 @@ pub trait ShieldedApi<U: ShieldedUtils + MaybeSend + MaybeSync>:
         }))
     }
 
+    /// Computes added_amt - required_amt taking care of denominations and asset
+    /// decodings. Error out if required_amt is not less than added_amt.
+    #[allow(async_fn_in_trait)]
+    async fn compute_change(
+        &mut self,
+        client: &(impl Client + Sync),
+        added_amt: I128Sum,
+        mut required_amt: ValueSum<(MaspDigitPos, Address), Change>,
```

Comment:
Could `ValueSum<(MaspDigitPos, Address), Change>` possibly be changed to `ValueSum<(MaspDigitPos, Address), i128>` here?